lib/common/socket.c

```diff
@@ -798,12 +800,16 @@ static int on_async_resumption_new(SSL *ssl, SSL_SESSION *session)
     i2d_SSL_SESSION(session, &p);
 
     id = SSL_SESSION_get_id(session, &id_len);
-    resumption_new(h2o_iovec_init(id, id_len), data);
+    resumption_new(sock, h2o_iovec_init(id, id_len), data);
     return 0;
 }
 
 static void on_async_resumption_remove(SSL_CTX *ssl_ctx, SSL_SESSION *session)
 {
+    if (resumption_remove == NULL) {
+        return;
+    }
```

Comment:
Do you recall why the remove callbacks were abolished in edecc08 (for redis) and a5a433b (for memcached)?

If there is some reason to not remove the session tickets, I think it would make more sense for us to make `on_async_resumption_remove` a no-op rather than relying on `resumption_remove == NULL`.